### PR TITLE
TS code examples

### PR DESCRIPTION
**Pull request by sequba**
### Context

- [x] convert all JS examples to TS by hand
  - [x] accessibility/
  - [x] cell-features/
  - [x] cell-types/
  - [x] formulas/
  - [x] navigation/
  - [x] rows/
  - [x] tools-and-building/
  - [x] accessories-and-menus/
  - [x] cell-functions/
  - [x] columns/
  - [x] getting-started/
  - [x] internationalization/
  - [x] optimization/
  - [x] security/
  - [x] technical-specification/
  - [x] upgrade-and-migration/
- [x] create a script that
  - compiles all TS examples files to JS (tsc)
  - formats them to be more human-readable (eslint)
- [x] run the script during the docs build

There are 192 js files in the `docs/content/guides` dir:
```sh
$ find docs/content/guides -name "*.js" | wc
    192     192   13076
```

### How has this been tested?

❗To be described by @magierg 

> - [ ] Verify that all JS examples run without errors
> - [ ] Verify that all TS examples run without errors
> - [ ] Verify that all TS examples produce no type errors

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. https://github.com/handsontable/dev-handsontable/issues/1867

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [ ] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

[skip changelog]

### CODE REVIEW

**Review by evanSe — APPROVED** [review]
In general everything looks good, a few improvements I would create.
- generate multiple examples files at once (currently it takes along time)
- do have generation as part of start script if its so slow. 


If speeding up the script takes more than an hour I wouldn't bother with it

**Comment by evanSe** [comment]
Also its worth rebasing this branch, it has 165 commits currently.

File: docs/.vuepress/tools/generate-js-examples.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+ #!/usr/bin/bash
```
[the remark this answers is not in the document]
Comment (sequba) [reply docs/.vuepress/tools/generate-js-examples.sh:1]: I changed it so it processes 16 files concurently.

File: docs/content/guides/cell-functions/cell-renderer/javascript/example5.ts [hunk 2]
```diff
@@ -1,19 +1,20 @@
 import Handsontable from 'handsontable';
 import 'handsontable/dist/handsontable.full.min.css';
+import { BaseRenderer } from 'handsontable/renderers';
 
 let isChecked = false;
-const exampleContainer = document.querySelector('#exampleContainer5');
-const container = document.querySelector('#example5');
+const exampleContainer = document.querySelector('#exampleContainer5')!;
```
[the remark this answers is not in the document]
Comment (sequba) [reply docs/content/guides/cell-functions/cell-renderer/javascript/example5.ts:6]: Exactly. I didn't want to introduce the null checking to keep the examples brief and clean.

File: docs/package.json [hunk 3]
```diff
@@ -14,9 +14,10 @@
     "docs:lint:fix": "eslint --ext .js,.vue,.mjs .vuepress content --fix",
     "docs:scripts:link-assets": "node .vuepress/tools/link-assets.mjs",
     "docs:scripts:generate-legacy-docs-versions": "node .vuepress/tools/generate-legacy-docs-versions.mjs",
-    "docs:start": "npm run docs:scripts:link-assets && cross-env NODE_OPTIONS=--openssl-legacy-provider node ./node_modules/.bin/vuepress dev ./",
-    "docs:start:no-cache": "npm run docs:scripts:link-assets && cross-env NODE_OPTIONS=--openssl-legacy-provider node ./node_modules/.bin/vuepress dev --no-cache ./",
-    "docs:build": "cross-env NODE_OPTIONS=--openssl-legacy-provider node .vuepress/tools/build.mjs -- --no-cache",
+    "docs:scripts:generate-js-examples": "bash .vuepress/tools/generate-js-examples.sh",
+    "docs:start": "npm run docs:scripts:link-assets && npm run docs:scripts:generate-js-examples && cross-env NODE_OPTIONS=--openssl-legacy-provider node ./node_modules/.bin/vuepress dev ./",
```
[the remark this answers is not in the document]
Comment (sequba) [reply docs/package.json]: Removed

File: docs/content/guides/cell-functions/cell-renderer/javascript/example5.ts [hunk 4]
```diff
@@ -22,28 +23,24 @@ const hot = new Handsontable(container, {
...
+      : `Some <input type="checkbox" class="checker" ${isChecked ? 'checked="checked"' : ''}> checkbox`;
   },
   autoWrapRow: true,
   autoWrapCol: true,
   licenseKey: 'non-commercial-and-evaluation'
 });
 
 exampleContainer.addEventListener('mousedown', (event) => {
-  if (event.target.nodeName == 'INPUT' && event.target.className == 'checker') {
+  if ((event.target as HTMLElement).nodeName == 'INPUT' && (event.target as HTMLElement).className == 'checker') {
```
[the remark this answers is not in the document]
Comment (sequba) [reply docs/content/guides/cell-functions/cell-renderer/javascript/example5.ts:36]: I couldn't find a way to avoid `as` in this situation.

**Comment by sequba** [comment]
Rebased. @evanSe please re-review.

**Review by evanSe — DISMISSED** [review]

**Comment by sequba** [comment]
@evanSe wrote [here](https://github.com/handsontable/handsontable/pull/10990#pullrequestreview-2070927317):

> do have generation as part of start script if its so slow.

@magierg wrote [here](https://github.com/handsontable/handsontable/pull/10990#issuecomment-2151564122):

> There is a problem when running `npm run docs:start` as TS -> JS conversion not being triggered

Currently `npm run docs:start` don't run `generate-js-examples` script, but it could be added there. Personally, I prefer the `start` script to be fast as we use it frequently during the development. @evanSe please, make a decision about it.

**Review by evanSe — APPROVED** [review]

**Pull request merged**